spec/controllers/tasks_controller_spec.rb

```diff
@@ -252,6 +278,28 @@
         expect { put_update }.to change { task.reload.license.name }.to('new_license')
       end
 
+      it 'updates the tasks labels' do
+        expect { put_update }.to change { task.reload.labels }.from([existing_label]).to([new_label])
+      end
+
+      context 'when requesting new label to be created' do
+        let(:changed_attributes) { {label_names: [not_existing_label_name]} }
+        let(:not_existing_label_name) { 'some new label' }
+
+        it 'creates new task label' do
+          expect { put_update }.to change(Label, :count).by(1)
+        end
+
+        it 'creates a label with the correct name' do
+          put_update
+          expect(Label.last.name).to eql not_existing_label_name
```

Comment:
I am still not a fan of the `Label.last` usage in tests. What if we search for the label by its name and expect that only one result is returned? Something like:

```
          expect(Label.where(name: not_existing_label_name).size).to be 1
```

Reply:
Thats probably better.